src/components/organisms/UiList/_internal/UiListItem.vue

```diff
@@ -3,32 +3,133 @@
     :is="tag"
     class="ui-list-item"
   >
-    <!-- @slot Use this slot to place list item content. -->
+    <!-- @slot Use this slot to place content inside list-item. -->
     <slot />
+    <!-- @slot Use this slot to replace suffix template -->
+    <slot
+      name="suffix"
+      v-bind="{
+        hasSuffix,
+        suffixAttrs: defaultProps.suffixAttrs
+      }"
+    >
+      <UiListItemSuffix
+        v-if="hasSuffix"
+        v-bind="defaultProps.suffixAttrs"
+        class="ui-list-item__suffix"
+      />
+    </slot>
   </component>
 </template>
 
 <script setup lang="ts">
+import { computed } from 'vue';
 import type { PropType } from 'vue';
 import type { HTMLTag } from '../../../../types/tag';
+import type { Icon } from '../../../../types/icon';
+import type { PropsAttrs } from '../../../../types/attrs';
+import UiListItemSuffix from './UiListItemSuffix.vue';
 
-defineProps({
+const props = defineProps({
```

Comment:
What do you think about **[this](https://vuejs.org/api/sfc-script-setup.html#typescript-only-features)** pattern with `withDefaults`? Wouldn't it be cleaner than all of those casts `as PropType<HTMLTag>,`, `as PropType<Icon>,`, `Object as PropsAttrs,` etc? :) 
+ It seems like vue3 docs suggest to rather use `withDefaults` 🤔

Reply:
oh cool 😀 earlier it was probably experimental. I would try this pattern. What do you think @pspaczek?